graphsdk/src/main/java/com/microsoft/graph/model/Date.java

```diff
@@ -0,0 +1,90 @@
+package com.microsoft.graph.model;
+
+
+import java.util.Locale;
+
+/**
+ * A timezone-nonspecific date
+ */
+public class Date {
+
+    /**
+     * Array indices for date parsing
+     */
+    private static final int
+            sYear = 0,
+            sMonth = 1,
+            sDay = 2;
+
+    /**
+     * The raw string source of this object
+     */
+    private final String rawDateString;
+
+    private final int
+            mYear,
+            mMonth,
+            mDay;
+
+    /**
+     * Constructs a timezone-nonspecific <code>Date</code>
+     *
+     * @param dateStr date string of the form <code>yyyy-mm-dd</code>
+     */
+    public Date(String dateStr) {
+        // save the String from which this object is derived
+        rawDateString = dateStr;
+
+        // break the date up into its constituent parts
+        String[] dateInfo = rawDateString.split("-");
```

Comment:
You should validate the length of the split string, if it is outside the expected bounds it should throw a parsing exception, otherwise we will throw a no-so-useful indexoutofbounds